packages/core/src/components/graphs/bar-stacked.ts

```diff
@@ -0,0 +1,217 @@
+						correspondingData[dataset.label] = 0;
+					}
+				});
+				correspondingData["label"] = date;
+
+				return correspondingData;
+			});
+		} else {
+			// Create the stack datalist
+			stackDataArray = displayData.labels.map((label, i) => {
+				const correspondingData = {};
+
+				displayData.datasets.forEach(dataset => {
+					correspondingData[dataset.label] = !isNaN(dataset.data[i]) ? dataset.data[i] : dataset.data[i].value;
+				});
+				correspondingData["label"] = label;
+
+				return correspondingData;
+			});
+		}
+
+		return stackDataArray;
+	}
+
+	render(animate: boolean) {
+		// Chart options mixed with the internal configurations
+		const options = this.model.getOptions();
+
+		// Grab container SVG
+		const svg = this.getContainerSVG();
+
+		// Create the data and keys that'll be used by the stack layout
+		const displayData = this.model.getDisplayData();
+		const stackDataArray = this.getStackData();
+		const stackKeys = displayData.datasets.map(dataset => dataset.label);
+
+		// Update data on all bar groups
+		const barGroups = svg.selectAll("g.bars")
+			.data(stack().keys(stackKeys)(stackDataArray), d => d.key);
+
+		// Remove elements that need to be exited
+		// We need exit at the top here to make sure that
+		// Data filters are processed before entering new elements
+		// Or updating existing ones
+		barGroups.exit()
+			.attr("opacity", 0)
+			.remove();
+
+		// Add bar groups that need to be introduced
+		barGroups.enter()
+			.append("g")
+			.classed("bars", true);
+
+		// Update data on all bars
+		const bars = svg.selectAll("g.bars").selectAll("rect.bar")
+			.data(d => addLabelsAndValueToData(d), d => d.label);
+
+		// Remove bars that need to be removed
+		bars.exit()
+			.remove();
+
+		// Update styling and position on existing bars
+		// As well as bars that were just added
+		bars.enter()
+			.append("rect")
+			.merge(bars)
+				.classed("bar", true)
+				.attr("x", (d, i) => {
+					const barWidth = this.getBarWidth();
+					return this.services.axes.getXValue(d, i) - (barWidth / 2);
+				})
+				.attr("width", this.getBarWidth.bind(this))
+				.transition(this.services.transitions.getTransition("bar-update-enter", animate))
+				.attr("y", (d, i) => this.services.axes.getYValue(d[1], i))
+				.attr("fill", d => this.model.getFillScale()[d.datasetLabel](d.label))
+				.attr("height", (d, i) => {
+					return this.services.axes.getYValue(d[0]) - this.services.axes.getYValue(d[1]);
+				})
+				.attr("opacity", 1);
+
+		// Add event listeners for the above elements
+		this.addEventListeners();
+	}
+
+	// Highlight elements that match the hovered legend item
+	handleLegendOnHover = e => {
+		const { hoveredElement } = e.detail;
+
+		this.parent.selectAll("rect.bar")
+			.transition(this.services.transitions.getTransition("legend-hover-bar"))
+			.attr("opacity", d => (d.datasetLabel !== hoveredElement.datum()["key"]) ? 0.3 : 1);
+	}
+
+	// Un-highlight all elements
+	handleLegendMouseOut = e => {
```

Comment:
```suggestion
	handleLegendMouseOut = event => {
```

Ditto for `handleLegendOnHover` ... is there a type we can give `event`? `MouseEvent` maybe?